src/Compilers/CSharp/Test/Emit2/Semantics/PatternMatchingTests_ListPatterns.cs

```diff
@@ -8915,4 +8915,211 @@ public void NotExhaustive_LongList()
             // _ = a switch { { Length: < 1000 } => 0 };
             Diagnostic(ErrorCode.WRN_SwitchExpressionNotExhaustive, "switch").WithArguments("{ Length: 1000 }").WithLocation(2, 7));
     }
+
+    [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/71660")]
+    public void MixedCountPatterns()
+    {
+        // One of the property patterns is treated as a non-negative Count pattern,
+        // while the other is a regular property pattern.
+        var source = """
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+System.Console.Write(select(new List<int>()));
+System.Console.Write(select(new List<int>() { 42 }));
+System.Console.Write(select(new C()));
+
+int select(ICollection c)
+{
+    try
+    {
+        return c switch
+        {
+            { Count: 0 } => 0,
+            IList { Count: > 0 } => 1,
+        };
+    }
+    catch
+    {
+        return 2;
+    }
+}
+
+class C : System.Collections.ICollection
+{
+    public int Count => 1;
+    public object SyncRoot => throw new NotImplementedException();
+    public bool IsSynchronized => throw new NotImplementedException();
+    public void CopyTo(Array array, int index) => throw new NotImplementedException();
+    public IEnumerator GetEnumerator() => throw new NotImplementedException();
+}
+""";
+        CompileAndVerify(source, expectedOutput: "012").VerifyDiagnostics(
+            // (13,18): warning CS8509: The switch expression does not handle all possible values of its input type (it is not exhaustive). For example, the pattern '{ Count: 1 }' is not covered.
```

Comment:
>// (13,18): warning CS8509: The switch expression does not handle all possible values of its input type (it is not exhaustive). For example, the pattern '{ Count: 1 }' is not covered. [](http://example.com/codeflow?start=12&length=183)

This warning looks unexpected to me. Both ```Count``` references refer to the same ```ICollection.Count``` property on the same instance. We checked it for 0 and for values >0, therefore '{ Count: 1 }' is also covered.

Reply:
We only checked for values >0 when the type is IList, so not for the general case.

Reply:
> We only checked for values >0 when the type is IList, so not for the general case.

Yes, I missed the fact. Thanks.